### Appriate x and y labels for each plot
### Comment on what each plot is showing
### Comment your code

Modules

In [1]:
import pandas as pd
pd.set_option("display.max_columns", None)
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

Load Data

In [2]:
df = pd.read_csv("Speed+Dating+Data.csv", encoding="unicode_escape")

In [29]:
df.shape

(8378, 195)

Descriptive Analysis

In [30]:
df_description = df.describe()
df_description

iid           id       gender          idg       condtn  \
count  8378.000000  8377.000000  8378.000000  8378.000000  8378.000000   
mean    283.675937     8.960248     0.500597    17.327166     1.828837   
std     158.583367     5.491329     0.500029    10.940735     0.376673   
min       1.000000     1.000000     0.000000     1.000000     1.000000   
25%     154.000000     4.000000     0.000000     8.000000     2.000000   
50%     281.000000     8.000000     1.000000    16.000000     2.000000   
75%     407.000000    13.000000     1.000000    26.000000     2.000000   
max     552.000000    22.000000     1.000000    44.000000     2.000000   

              wave        round     position     positin1        order  \
count  8378.000000  8378.000000  8378.000000  6532.000000  8378.000000   
mean     11.350919    16.872046     9.042731     9.295775     8.927668   
std       5.995903     4.358458     5.514939     5.650199     5.477009   
min       1.000000     5.000000     1.000000     1.000000     1.000000   
25%       7.000000    14.000000     4.000000     4.000000     4.000000   
50%      11.000000    18.000000     8.000000     9.000000     8.000000   
75%      15.000000    20.000000    13.000000    14.000000    13.000000   
max      21.000000    22.000000    22.000000    22.000000    22.000000   

           partner          pid        match     int_corr     samerace  \
count  8378.000000  8368.000000  8378.000000  8220.000000  8378.000000   
mean      8.963595   283.863767     0.164717     0.196010     0.395799   
std       5.491068   158.584899     0.370947     0.303539     0.489051   
min       1.000000     1.000000     0.000000    -0.830000     0.000000   
25%       4.000000   154.000000     0.000000    -0.020000     0.000000   
50%       8.000000   281.000000     0.000000     0.210000     0.000000   
75%      13.000000   408.000000     0.000000     0.430000     1.000000   
max      22.000000   552.000000     1.000000     0.910000     1.000000   

             age_o       race_o     pf_o_att     pf_o_sin     pf_o_int  \
count  8274.000000  8305.000000  8289.000000  8289.000000  8289.000000   
mean     26.364999     2.756653    22.495347    17.396867    20.270759   
std       3.563648     1.230689    12.569802     7.044003     6.782895   
min      18.000000     1.000000     0.000000     0.000000     0.000000   
25%      24.000000     2.000000    15.000000    15.000000    17.390000   
50%      26.000000     2.000000    20.000000    18.370000    20.000000   
75%      28.000000     4.000000    25.000000    20.000000    23.810000   
max      55.000000     6.000000   100.000000    60.000000    50.000000   

          pf_o_fun     pf_o_amb     pf_o_sha        dec_o       attr_o  \
count  8280.000000  8271.000000  8249.000000  8378.000000  8166.000000   
mean     17.459714    10.685375    11.845930     0.419551     6.190411   
std       6.085526     6.126544     6.362746     0.493515     1.950305   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%      15.000000     5.000000     9.520000     0.000000     5.000000   
50%      18.000000    10.000000    10.640000     0.000000     6.000000   
75%      20.000000    15.000000    16.000000     1.000000     8.000000   
max      50.000000    53.000000    30.000000     1.000000    10.500000   

            sinc_o      intel_o        fun_o        amb_o       shar_o  \
count  8091.000000  8072.000000  8018.000000  7656.000000  7302.000000   
mean      7.175256     7.369301     6.400599     6.778409     5.474870   
std       1.740575     1.550501     1.954078     1.794080     2.156163   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       6.000000     6.000000     5.000000     6.000000     4.000000   
50%       7.000000     7.000000     7.000000     7.000000     6.000000   
75%       8.000000     8.000000     8.000000     8.000000     7.000000   
max      10.000000    10.000000    11.000000    10.000000    10.000000   

            l

In [31]:
df_description["gender"]["mean"].round(2)

0.5

Throught this descriptive analysis, we can see that that we have approximately the same number of men and women in our dataset. The dataset is balanced.

1. What is the age range of the participants?

In [128]:
df["gender"] = df["gender"].map({0: "Female", 1: "Male"})

In [33]:
fig_age = px.box(
    df[["age", "gender"]],
    y="gender",
    x="age",
    template="simple_white",
    color="gender",
    width=800,
    
)
fig_age.update_layout(title_text="Age Distribution")
fig_age.update(layout=dict(title=dict(x=0.5)))
fig_age.show()

We can see that the participants are mostly young beacause there are between 21 and 30 years old. And men are mostly 

2. Which gender most validates the other gender.

In [34]:
df["match"] = df["match"].map({0: "No", 1: "Yes"})
df["dec"] = df["dec"].map({0: "No", 1: "Yes"})

In [35]:
fig_match = px.histogram(
    df, x="dec", color="gender", template="simple_white", width=700, barmode="group"
)
fig_match.update_layout(title_text="Gender Decision")
fig_match.update(layout=dict(title=dict(x=0.5)))
fig_match.show()

Men validate most the female partners.

3. How many matches are there between the participants?

In [36]:
fig_match = px.pie(df["match"].value_counts().reset_index(), values="count", color = "match", template="simple_white", width=700)
fig_match.update_layout(title_text="Matches")
fig_match.update(layout=dict(title=dict(x=0.5)))
fig_match.update_traces(pull = [0, 0.1])
fig_match.show()

In [37]:
df_match = df["match"].value_counts().reset_index()
df_match["count"][0] / df_match["count"][1]

5.071014492753624

That ratio suggests that, there are five times more individuals who are incompatibles for a second date compared to those who mutually agree on pursuing a second date.

4. What is the main goal of the participants?

In [38]:
dict_goal = {
    1: "Seemed like a fun night out",
    2: "To meet new people",
    3: "To get a date",
    4: "Looking for a serious relationship",
    5: "To say I did it",
    6: "Other",
}
df_goal = df["goal"].map(dict_goal).value_counts(normalize=True).reset_index()
df_goal

goal  proportion
0         Seemed like a fun night out    0.412821
1                  To meet new people    0.362935
2                       To get a date    0.076033
3                     To say I did it    0.061453
4                               Other    0.050488
5  Looking for a serious relationship    0.036269

In [39]:
df_goal_gender = df.loc[:,["goal", "gender"]]
df_goal_gender["goal"] = df_goal_gender["goal"].map(dict_goal)

In [40]:
fig_goal_gender = px.sunburst(
    df_goal_gender.dropna(),
    path=["goal", "gender"],
    template="simple_white",
    width=700,
)
fig_goal_gender.update_layout(title_text="Main Goal by Gender")
fig_goal_gender.update(layout=dict(title=dict(x=0.5)))
fig_goal_gender.show()

We can see that the objectives of speed dating are primarily shallow for most participants `like a fun night out`, `to meet new people`. Only 3.63% of the participants are looking for a serious relationship. 
Men are more interested by `get a date` and `looking for a serious relationship` than women.

5. What is the number of compatible individuals per participant goal?

In [41]:
df_goal_match = df[["goal", "match"]].value_counts().reset_index()
df_goal_match.rename(columns={"count": "match_count"}, inplace=True)

total_goal = df["goal"].value_counts().reset_index()

df_goal_match = df_goal_match.merge(total_goal, on="goal", how="left")
df_goal_match["match_rate"] = df_goal_match["match_count"] / df_goal_match["count"] *100
df_goal_match["goal"] = df_goal_match["goal"].map(dict_goal)
df_goal_match

goal match  match_count  count  match_rate
0          Seemed like a fun night out    No         2843   3426   82.983071
1                   To meet new people    No         2528   3012   83.930943
2          Seemed like a fun night out   Yes          583   3426   17.016929
3                        To get a date    No          531    631   84.152139
4                   To meet new people   Yes          484   3012   16.069057
5                      To say I did it    No          425    510   83.333333
6                                Other    No          357    419   85.202864
7   Looking for a serious relationship    No          250    301   83.056478
8                        To get a date   Yes          100    631   15.847861
9                      To say I did it   Yes           85    510   16.666667
10                               Other   Yes           62    419   14.797136
11  Looking for a serious relationship   Yes           51    301   16.943522

In [42]:
# Plot stacked bar chart
fig_goal_match = px.bar(
    df_goal_match[df_goal_match["match"]=="Yes"].sort_values(by="match_rate"),
    x="match_rate",
    y="goal",
    color="match",
    labels={"goal": "Goal"},
    width=800,
    orientation="h"
    
)
fig_goal_match.update_layout(title_text="Normalized Count of Matches by Goal")
fig_goal_match.update(layout=dict(title=dict(x=0.5)))
fig_goal_match.show()

Through this plot, we can see that there is no a big difference between the goals and the match rate.

6. Before the speed dating event, what are the least desirable attributes in a male partner? Does this differ for female partners?

In [43]:
# Compute the mean of the desirable attributes for each gender before speed dating event

df_desirable_attr = (
    df[["gender", "attr1_1", "sinc1_1", "intel1_1", "fun1_1", "amb1_1", "shar1_1"]]
    .groupby("gender")
    .mean()
    .reset_index()
)
df_desirable_attr

gender    attr1_1    sinc1_1   intel1_1     fun1_1     amb1_1    shar1_1
0  Female  18.055224  18.305008  21.002502  17.147292  12.827222  12.704194
1    Male  26.921689  16.498436  19.537374  17.763893   8.552829  10.996574

In [44]:
# Create a dictionary to map the attribute names to a more descriptive name

dict_desirable_attr = {
    "gender": "Gender",
    "attr1_1": "Attractive",
    "sinc1_1": "Sincere",
    "intel1_1": "Intelligent",
    "fun1_1": "Fun",
    "amb1_1": "Ambitious",
    "shar1_1": "Shared Interests",
}

# Map the attribute names to the dictionary
df_desirable_attr.columns = df_desirable_attr.columns.map(dict_desirable_attr)

In [45]:
def sort_values(values):
    return values.sort_values(ascending=False)

In [46]:
df_desirable_attr.iloc[0, :].res

Gender                 Female
Attractive          18.055224
Sincere             18.305008
Intelligent         21.002502
Fun                 17.147292
Ambitious           12.827222
Shared Interests    12.704194
Name: 0, dtype: object

In [ ]:
sort_values(df_desirable_attr.iloc[0])

In [148]:
# rows = 2
# cols = 3
# fig = make_subplots(rows=rows, cols=cols, shared_xaxes=True)
# row = 1
# col = 1
# for i in ["attr1_s", "sinc1_s", "intel1_s", "fun1_s", "amb1_s", "shar1_s"]:
#     fig.add_trace(
#         go.Bar(
#             x=df_desirable_attr["gender"],
#             y=df_desirable_attr[i],
#             name=dict_desirable_attr[i],
#         ),
#         row=row,
#         col=col,
#     )
#     row += 1
#     if row > rows:
#         row = 1
#         col += 1


# fig.update_layout(
#     title_text="Desirable Attributes",
#     template="simple_white",
#     width=800,
#     height=800,
#     barmode="group",
# )
# fig.update(layout=dict(title=dict(x=0.5)))
# fig.show()

In [54]:
df_desirable_attr["Gender"].index[0]

0

In [91]:
df_desirable_attr

Gender  Attractive    Sincere  Intelligent        Fun  Ambitious  \
0  Female   18.055224  18.305008    21.002502  17.147292  12.827222   
1    Male   26.921689  16.498436    19.537374  17.763893   8.552829   

   Shared Interests  
0         12.704194  
1         10.996574

In [109]:
df_desirable_attr[df_desirable_attr["Gender"] == "Male"].iloc[0][1:].sort_values(ascending=False).index

Index(['Attractive', 'Intelligent', 'Fun', 'Sincere', 'Shared Interests',
       'Ambitious'],
      dtype='object')

In [114]:
df_desirable_attr[df_desirable_attr["Gender"] == "Male"].iloc[0][1:]

Attractive          26.921689
Sincere             16.498436
Intelligent         19.537374
Fun                 17.763893
Ambitious            8.552829
Shared Interests    10.996574
Name: 1, dtype: object

In [111]:
# Initialize the rows and columns of the plot

rows = 1
cols = 2
fig_attr_desirable = make_subplots(rows=rows, cols=cols, shared_yaxes=True)

row = 1
col = 1

# Add the traces to the plot

for i in df_desirable_attr["Gender"].unique():
    fig_attr_desirable.add_trace(
        go.Bar(
            x=df_desirable_attr[df_desirable_attr["Gender"] == "Male"].iloc[0][1:].sort_values(ascending=False).index,
            y=df_desirable_attr[df_desirable_attr["Gender"] == i].iloc[0][1:].sort_values(ascending=False),
            name=i,
        ),
        row=row,
        col=col,
    )
    row = 1
    col += 1


fig_attr_desirable.update_layout(
    title_text="Desirable Attributes",
    template="simple_white",
    width=1400,
    height=500,
    barmode="group",
)
fig_attr_desirable.update(layout=dict(title=dict(x=0.5)))
fig_attr_desirable.show()

In [ ]:
# Initialize the rows and columns of the plot

rows = 1
cols = 2
fig_attr_desirable = make_subplots(rows=rows, cols=cols, shared_yaxes=True)

row = 1
col = 1

# Add the traces to the plot

for gender in df_desirable_attr["Gender"].unique():
    df_filtered = df_desirable_attr[
        df_desirable_attr["Gender"] == gender].T.reset_index()
    df_filtered.columns = ["characteristics", "mean"]
    df_filtered = df_filtered.drop(0, axis=0).sort_values(
        by="mean", ascending=False
    )
    fig_attr_desirable.add_trace(
        go.Bar(
            x=df_filtered["characteristics"],
            y=df_filtered["mean"],
            name=gender,
        ),
        row=row,
        col=col,
    )
    row = 1
    col += 1


fig_attr_desirable.update_layout(
    title_text="Desirable Attributes",
    template="simple_white",
    width=1400,
    height=500,
    barmode="group",
)
fig_attr_desirable.update(layout=dict(title=dict(x=0.5)))
fig_attr_desirable.show()


We can infer that among the attributes of `Attractive, Sincere, Intelligent, Fun, and Shared Interests`, the least desirable one for both genders in the experiment is `Ambitious`. The most desirable attribute for men is `Attractive`, while for women it is `Intelligent`.

7. How important do people think attributes is in potential mate selection vs. its real impact?

In [117]:
df[["attr7_2", "gender"]][(df["match"] == "Yes") & (df["gender"] == "Male")].value_counts().sort_index().reset_index()

attr7_2  gender
30.0     Male      39
50.0     Male      31
40.0     Male      21
70.0     Male      15
20.0     Male      12
25.0     Male       9
10.0     Male       9
22.0     Male       9
60.0     Male       5
18.0     Male       3
35.0     Male       2
80.0     Male       2
Name: count, dtype: int64

In [150]:
fig_attrib_match = make_subplots(
    rows=6, cols=2, subplot_titles=("Female Features", "Male Features")
)


# Define a function to generate traces
def add_trace(attribute, row, col, color, gender):
    numerator = (
        df[[attribute, "gender"]][(df["match"] == "Yes") & (df["gender"] == gender)]
        .value_counts()
        .sort_index()
        .reset_index()
    )
    denominator = (
        df[[attribute, "gender"]][(df["gender"] == gender)]
        .value_counts()
        .sort_index()
        .reset_index()
    )
    result = numerator[["count", attribute]].merge(
        denominator, how="left", left_on=attribute, right_on=attribute
    )
    result["rate"] = result["count_x"] / result["count_y"] * 100
    fig_attrib_match.add_trace(
        go.Scatter(
            x=result[attribute],
            y=result["rate"],
            line=dict(color=color, width=1),
            mode="lines+markers",
        ),
        row=row,
        col=col,
    )


# Attactiveness rated for Men
add_trace("attr7_2", 1, 2, "orange", "Male")

# Sincerity rated for Men
add_trace("sinc7_2", 2, 2, "orange", "Male")

# Intelligent rated for Men
add_trace("intel7_2", 3, 2, "orange", "Male")

# Fun rated for Men
add_trace("fun7_2", 4, 2, "orange", "Male")

# Ambition rated for Men
add_trace("amb7_2", 5, 2, "orange", "Male")

# Shared Interest rated for Men
add_trace("shar7_2", 6, 2, "orange", "Male")

# Attactiveness rated for Women
add_trace("attr7_2", 1, 1, "royalblue", "Female")

# Sincerity rated for Women
add_trace("sinc7_2", 2, 1, "royalblue", "Female")

# Intelligent rated for Women
add_trace("intel7_2", 3, 1, "royalblue", "Female")

# Fun rated for Women
add_trace("fun7_2", 4, 1, "royalblue", "Female")

# Ambition rated for Women
add_trace("amb7_2", 5, 1, "royalblue", "Female")

# Shared Interest rated for Women
add_trace("shar7_2", 6, 1, "royalblue", "Female")


fig_attrib_match.update_layout(
    xaxis1_title="Mark",
    xaxis2_title="Mark",
    xaxis3_title="Mark",
    xaxis4_title="Mark",
    xaxis5_title="Mark",
    xaxis6_title="Mark",
    xaxis7_title="Mark",
    xaxis8_title="Mark",
    xaxis9_title="Mark",
    xaxis10_title="Mark",
    xaxis11_title="Mark",
    xaxis12_title="Mark",
    yaxis1_title="% Attractiveness",
    yaxis2_title="% Attractiveness",
    yaxis3_title="% Sincerity",
    yaxis4_title="% Sincerity",
    yaxis5_title="% Intelligence",
    yaxis6_title="% Intelligence",
    yaxis7_title="% Fun",
    yaxis8_title="% Fun",
    yaxis9_title="% Ambitious",
    yaxis10_title="% Ambitious",
    yaxis11_title="% Sharing of Interest",
    yaxis12_title="% Sharing of Interest",
    legend_tracegroupgap=1000,
    showlegend=False,
    title={
        "text": "Mark depending on the features rating",
        "y": 0.97,
        "x": 0.5,
        "xanchor": "center",
        "yanchor": "top",
    },
    height=1200,
    width=850,
)

fig_attrib_match.show()

* This graphs show that attractiveness is very important in potential mate selection for both men and women based on the mark range of each (0 to 80 for men and 0 to greather than 65 for women).
* For women, the important criteria are very variable but the marked ones are sincerity and ambitious
* One more interesting thing to notice is that is not the women with the most attractive characteristics who are selected as potential mate, but the women  with the most intelligent characteristics.
* For both genders, the more shared interests they have, the less likely they are to be selected as a potential mate.

8. Are religion more important than a shared racial background?

In [151]:
df_reigion_men = (
    (
        df[["imprelig"]][(df["dec"] == "Yes") & (df["gender"] == "Male")]
        .value_counts()
        .sort_index()
        / df[["imprelig"]][(df["gender"] == "Male")].value_counts().sort_index()
        * 100
    )
    .reset_index()
    .rename(columns={"count": "reli_count"})
)
df_race_men = (
    (
        df[["imprace"]][(df["dec"] == "Yes") & (df["gender"] == "Male")]
        .value_counts()
        .sort_index()
        / df[["imprace"]][(df["gender"] == "Male")].value_counts().sort_index()
        * 100
    )
    .reset_index()
    .rename(columns={"count": "race_count"})
)
df_reli_race_men = df_race_men.merge(
    df_reigion_men, how="left", right_on="imprelig", left_on="imprace"
)

df_reigion_women = (
    (
        df[["imprelig"]][(df["dec"] == "Yes") & (df["gender"] == "Female")]
        .value_counts()
        .sort_index()
        / df[["imprelig"]][(df["gender"] == "Female")].value_counts().sort_index()
        * 100
    )
    .reset_index()
    .rename(columns={"count": "reli_count"})
)
df_race_women = (
    (
        df[["imprace"]][(df["dec"] == "Yes") & (df["gender"] == "Female")]
        .value_counts()
        .sort_index()
        / df[["imprace"]][(df["gender"] == "Female")].value_counts().sort_index()
        * 100
    )
    .reset_index()
    .rename(columns={"count": "race_count"})
)
df_reli_race_women = df_reigion_women.merge(
    df_race_women,
    how="left",
    left_on="imprelig",
    right_on="imprace",
)

In [152]:
fig_religion_race = make_subplots(rows=1, cols=2, subplot_titles=( "Women","Men"))


fig_religion_race.add_trace(
    go.Scatter(
        x=df_reli_race_women["imprace"],
        y=df_reli_race_women["reli_count"],
        line=dict(color="orange", width=1),
        mode="lines+markers",
        showlegend=False,
    ),
    row=1,
    col=1,
)

fig_religion_race.add_trace(
    go.Scatter(
        x=df_reli_race_women["imprace"],
        y=df_reli_race_women["race_count"],
        line=dict(color="royalblue", width=1),
        mode="lines+markers",
        showlegend=False,
    ),
    row=1,
    col=1,
)

fig_religion_race.add_trace(
    go.Scatter(
        x=df_reli_race_men["imprace"],
        y=df_reli_race_men["reli_count"],
        line=dict(color="orange", width=1),
        mode="lines+markers",
        name="Religion"
    ),
    row=1,
    col=2,
)

fig_religion_race.add_trace(
    go.Scatter(
        x=df_reli_race_men["imprace"],
        y=df_reli_race_men["race_count"],
        line=dict(color="royalblue", width=1),
        mode="lines+markers",
        name="Race"
    ),
    row=1,
    col=2,
)

fig_religion_race.update_layout(
    xaxis1_title="Scale",
    xaxis2_title="Scale",
    yaxis1_title=" Matches ratio",

    legend_tracegroupgap=1000,
    showlegend=True,
    title={
        "text": "Mark depending on the features rating",
        "y": 0.97,
        "x": 0.5,
        "xanchor": "center",
        "yanchor": "top",
    },
    height=500,
    width=1500,
)
fig_religion_race.show()

In these graphs, we observe that religion and race play a less significant role in women's selection of potential partners compared to men. For men, when a participant shares the same religion, they are more likely to receive a positive response. Conversely, as the participant's racial background becomes more similar to that of the man, her chances of receiving a positive opinion decrease.

9. Which field of study have more matching relationships ?

#### Check if the sample is balanced based on fields

In [127]:
fields_dict = {
    1: "Law",
    2: "Math",
    3: "Social Science, Psychologist",
    4: "Medical Science, Pharmaceuticals, and Bio Tech",
    5: "Engineering",
    6: "English/Creative Writing/Journalism",
    7: "History/Religion/Philosophy",
    8: "Business/Econ/Finance",
    9: "Education, Academia",
    10: "Biological Sciences/Chemistry/Physics",
    11: "Social Work",
    12: "Undergrad/undecided",
    13: "Political Science/International Affairs",
    14: "Film",
    15: "Fine Arts/Arts Administration",
    16: "Languages",
    17: "Architecture",
    18: "Other",
}

df["field_cd"] = df['field_cd'].map(fields_dict)
df["field_cd"].value_counts(normalize=True).reset_index()

field_cd  proportion
0                            Business/Econ/Finance    0.232040
1            Biological Sciences/Chemistry/Physics    0.119696
2                                      Engineering    0.104147
3          Political Science/International Affairs    0.085463
4                     Social Science, Psychologist    0.083896
5                                              Law    0.080159
6                              Education, Academia    0.075458
7                                      Social Work    0.056413
8              English/Creative Writing/Journalism    0.039176
9                      History/Religion/Philosophy    0.029050
10                                            Math    0.024952
11                   Fine Arts/Arts Administration    0.022541
12  Medical Science, Pharmaceuticals, and Bio Tech    0.017237
13                                            Film    0.015188
14                                           Other    0.006268
15                                       Languages    0.004822
16                             Undergrad/undecided    0.002290
17                                    Architecture    0.001205

This dataframe reveals an imbalance in the distribution of participants across different fields

In [162]:
df_field = df[["gender", "field_cd"]][df["match"] == "Yes"].value_counts().reset_index()
df_field_male = df_field[["field_cd", "count"]][df_field["gender"] == "Male"]
df_field_female = df_field[["field_cd", "count"]][df_field["gender"] == "Female"]

In [163]:
# Initialize the rows and columns of the plot
fig_field_desirable = make_subplots(rows=1, cols=2, shared_yaxes=True)

# Add the traces to the plot

fig_field_desirable.add_trace(
    go.Bar(
        x=df_field_female["field_cd"][:8],
        y=df_field_female["count"][:8],
        name="Female",
    ),
    row=row,
    col=col,
)
row = 1
col = 2

# Add the traces to the plot

fig_field_desirable.add_trace(
    go.Bar(x=df_field_male["field_cd"][:8], y=df_field_male["count"][:8], name="Male"),
    row=row,
    col=col,
)
row = 1
col = 1


fig_field_desirable.update_layout(
    title_text="Desirable Attributes",
    template="simple_white",
    width=1400,
    height=500,
    barmode="group",
)
fig_field_desirable.update(layout=dict(title=dict(x=0.5)))
fig_field_desirable.show()

Througth this graph we can notice that men who are in Business/Econ/Finance have more chances to get a second date than the order. In general, people who are in  Science have more chances to get a second date

10. In terms of getting a second date, is it better to be someone's first speed date of the night or their last?

In [141]:
contengence = pd.crosstab(df["order"], df["match"])
contengence

match    0    1
order          
1      424  127
2      463   88
3      460   91
4      452   99
5      454   97
6      453   88
7      435   79
8      429   79
9      401  101
10     390   84
11     311   61
12     326   46
13     315   57
14     318   54
15     290   44
16     263   41
17     224   29
18     213   40
19     145   33
20     118   26
21      80    6
22      34   10

In [142]:
from scipy.stats import chi2_contingency
chi2, p, dof, expected = chi2_contingency(contengence)

In [145]:
print(p)

0.0006942704805738717


In [175]:
df_order = df[["order", "match"]].value_counts().sort_index().reset_index()
df_order = df_order[df_order["match"] == "Yes"].merge(
    df["order"].value_counts().sort_index().reset_index(),
    left_on="order",
    right_on="order",
)
df_order['order_rate']= df_order["count_x"] / df_order["count_y"] *100
df_order

order match  count_x  count_y  order_rate
0       1   Yes      127      551   23.049002
1       2   Yes       88      551   15.970962
2       3   Yes       91      551   16.515426
3       4   Yes       99      551   17.967332
4       5   Yes       97      551   17.604356
5       6   Yes       88      541   16.266174
6       7   Yes       79      514   15.369650
7       8   Yes       79      508   15.551181
8       9   Yes      101      502   20.119522
9      10   Yes       84      474   17.721519
10     11   Yes       61      372   16.397849
11     12   Yes       46      372   12.365591
12     13   Yes       57      372   15.322581
13     14   Yes       54      372   14.516129
14     15   Yes       44      334   13.173653
15     16   Yes       41      304   13.486842
16     17   Yes       29      253   11.462451
17     18   Yes       40      253   15.810277
18     19   Yes       33      178   18.539326
19     20   Yes       26      144   18.055556
20     21   Yes        6       86    6.976744
21     22   Yes       10       44   22.727273

In [177]:
fig_order_match = px.bar(
    df_order,
    x="order",
    y="order_rate",
    width=800,
)
fig_order_match.update_layout(title_text="Match rate by order")
fig_order_match.update(layout=dict(title=dict(x=0.5)))
fig_order_match.show()

* First people have more chance of getting a match
* People who pass the test from 9 to 11 order have also more chance of getting a match which corresponds to the period when all the participants take a break
* based on the number of people who pass the in last position we cannot assume that be the last increase the chance of getting a match

#### Can people accurately predict their own perceived value in the dating market?

In [ ]:
df[i].isnull().value_counts(normalize=True).reset_index()["True"] > 0.95

In [19]:
df_own_appr = df[["iid","attr3_2", "sinc3_2", "intel3_2", "fun3_2", "amb3_2"]].groupby('iid').mean().reset_index()

In [20]:
df_own_appr["mean"] = df_own_appr.iloc[:, 1:].mean(skipna=True, axis=1)

In [21]:
df_own_appr_match = df[["iid", "dec_o"]].groupby("iid").sum().reset_index()

In [25]:
df_own_appr = df_own_appr.merge(
    df_own_appr_match, how="inner", left_on="iid", right_on="iid"
)

In [26]:
df_own_appr["mean"].corr(df_own_appr["dec_o"]) * 100

9.715903909708107

This means that there is almost no correlation between how participants rated themselves and the decisions made by their partners. If people have confidence in themselves and rate themselves highly, they should theoretically have a better chance of receiving a positive response from their partner. Conversely, the opposite would also be true.

However, it is important to remember that the main goal of the participants was not to find a serious relationship. Therefore, even if someone is confident and gives themselves a high rating, a partner who is participating just for fun might still give a negative decision for a second date.